llvmlite/binding/object_file.py

```diff
@@ -0,0 +1,59 @@
+from . import ffi
+from ctypes import c_bool, c_char_p, c_size_t, c_ulonglong, string_at
+
+class SectionIteratorRef(ffi.ObjectRef):
+    def name(self):
+        return ffi.lib.LLVMPY_GetSectionName(self)
+    def is_text(self):
+        return ffi.lib.LLVMPY_IsSectionText(self)
+    def size(self):
+        return ffi.lib.LLVMPY_GetSectionSize(self)
+    def data(self):
+        return string_at(ffi.lib.LLVMPY_GetSectionContents(self), self.size())
+    def is_end(self, object_file):
+        return ffi.lib.LLVMPY_IsSectionIteratorAtEnd(object_file, self)
+    def next(self):
```

Comment:
that's what is done below in the ObjectFileRef object. I did it this way to keep SectionIteratorRef close to the LLVM C API, and use ObjectFileRef as a wrapper for this.
Note that one issue with this original API is that previous iterators are invalid at each iterator, which is something that someone in python wouldn't expect.
For instance, if someone do `sections = list(ObjectFileRef().sections())`, it will get a list with only the last section.